### PR TITLE
#4: Set up Travis-CI

### PR DESCRIPTION
**Pull request by imjalpreet**
I have added the .travis.yml file as requested by you and it is based on the .travis.yml present in the gallery app.

### CODE REVIEW

**Comment by imjalpreet** [comment]
@oparoz I wanted to know what all is not required right now. I think the acceptance tests part is not required yet, so should I include it or remove it for now?

File: .travis.yml [hunk 1]
```diff
@@ -1,53 +1,140 @@
...
 env:
-    global:
-        - CORE_BRANCH=stable9
-    matrix:
-        - DB=pgsql
+  global:
+    - CORE_BRANCH=master
+    - APP_NAME=mediametadata
+    - GH_REF: github.com/interfasys/mediametadata.git
+    - secure: "cLmMqrbVZrvn2GAoVs6JWa5h70tzrAISjPZUfENjl0KYT6B84z3B94W/dGU+I2mxwvyBqEP+x83Bih2CxEflqc7tJmdqC+MM/ZZiigPX7RCjq5XHbzUpPpMbC8MFxT2K3jcFXUitfwZvNaZXJIiK3ZQJU4ayKaegLvI91x1SqH0="
```
Comment (imjalpreet) [inline .travis.yml]: @oparoz Can you tell what is this code? How do I generate it for my app?

Comment (oparoz) [inline .travis.yml]: You can remove GH_REF and secure. Those are needed by some external tools and when generating documentation

File: .travis.yml [hunk 2]
```diff
@@ -1,53 +1,140 @@
...
+  # Send coverage report to Scrutinizer
+  - sh -c "if [ '$TRAVIS_PHP_VERSION' = '7.0' ] && [ '$DB' = 'mysql' ]; then wget https://scrutinizer-ci.com/ocular.phar; fi"
+  - sh -c "if [ '$TRAVIS_PHP_VERSION' = '7.0' ] && [ '$DB' = 'mysql' ]; then php ocular.phar code-coverage:upload --format=php-clover tests/_output/coverage.xml; fi"
+
+  # Generate API documentation
+  - sh -c "if [ '$TRAVIS_PHP_VERSION' = '7.0' ] && [ '$DB' = 'mysql' ]; then php vendor/bin/phpdoc run; fi"
+  # Generate Wiki documentation
+  - sh -c "if [ '$TRAVIS_PHP_VERSION' = '7.0' ] && [ '$DB' = 'mysql' ]; then git submodule update --init; fi"
+  # Send documentation to Github Pages
+  - sh -c "if [ '$TRAVIS_PHP_VERSION' = '7.0' ] && [ '$DB' = 'mysql' ]; then cd build/documentation; bash ./docpublisher.sh; fi"
```
Comment (oparoz) [inline .travis.yml]: Please remove

Comment (oparoz) [inline .travis.yml]: You'll need to install the publishing script

File: .travis.yml [hunk 3]
```diff
@@ -1,53 +1,140 @@
...
+  - composer show -i
 
 script:
-    - make test
\ No newline at end of file
+  # Test lint
+  - php vendor/bin/parallel-lint --exclude vendor/composer/autoload_static.php --exclude travis --exclude vendor . vendor/composer vendor/symfony/yaml vendor/autoload.php
+
+  # Acceptance tests, using Firefox, without code coverage
+  - php vendor/bin/codecept run acceptance --env firefox
```
Comment (oparoz) [inline .travis.yml]: You can comment it out for now

**Comment by oparoz** [comment]
In general, you need to modify the codeception commands to only run unit tests for now

File: .travis.yml (hunk 2 above)
Comment (oparoz) [inline .travis.yml]: Actually, just comment things out for now and I'll re-enable that later

**Comment by oparoz** [comment]
Don't forget to set labels, milestone, etc and to ping me when you need a review.
I just took a look and things seem OK, but let me know if you still need to work on it.
In the PMS, the associated tasks is still at "Ready" btw.

**Comment by oparoz** [comment]
File looks OK, so merging and we will fix things later if there are problems with tests.

**Pull request merged**